### ISSUE TITLE
[Book 1, Chapter 12.2] Image rendered is flipped vertically

### ISSUE THREAD

**Issue by Shayan1503** [RayTracing/raytracing.github.io#1579]
When I render the image with the code provided in the book, I get an image which is flipped vertically compared to the image given in the book. I have discovered the error to be on line 86 of [camera.h](https://github.com/RayTracing/raytracing.github.io/blob/release/src/InOneWeekend/camera.h) of the first book. The particular line I am referring to has the following code: `vec3 viewport_u = viewport_width * u;`. 

The image is rendered as shown in the book with a simple fix:
```
vec3 viewport_u = viewport_width * -u;
```

I have cross-checked my code with the code in the book several times and it seems like the problem is with the code in the book. I might be wrong though, someone please let me know.

**Comment by hollasch** [comment]
`viewport_u` is related to the _horizontal_ of the rendered image. Something seems wrong here. As of v4.0.0-alpha.2, we've verified each step listing-by-listing (we call these "progressions"). Check out branch `progression/1/2024-0404` for the latest progression, particularly commit `973bbf5b` (Listing 83 Positionable and orientable camera). You can compare this state with your own code to see if there's a critical difference.

**Comment by Shayan1503** [comment]
There is a mistake in the way I have worded my problem. What I meant to say was that the image rendered were flipped on the _vertical axis_ or in other words flipped _horizontally_. I apologize for this stupid mistake.
After thorough cross-checking I have discovered that the mistake was on my part. I had the order of the parameters in the `cross` multiplication function of `vec3.h` in the wrong order, which was resulting in the images being flipped horizontally.
